### ISSUE TITLE
Question about Adult Dataset preprocessing

### ISSUE THREAD

**Issue by rrmina** [eth-sri/lcifr#1]
Hi! First of all thanks for sharing your code.

So I have looked at your data preprocessing codes, adult dataset specifically and I noticed that the sensitive/protected variable('sex') is also included in both your train_features and test_features.

My question is, was this intended i.e. (did you intend your algorithm to have access on both features and sensitive variables, both at train time and test time)?

Thank you very much.

**Comment by anianruoss** [comment]
Thank you for your interest in our work.

In the framework that we build upon ([McNamara et al.](https://arxiv.org/abs/1710.04394)), the data producer has access to all features of the dataset. The data producer then tries to learn a latent representation where similar individuals are mapped at most epsilon apart in l-infinity distance. The data consumer who builds the downstream application has access only to this latent representation.

To see why this latent representation is independent of the protected attribute sex, consider the constraint that two individuals should be regarded as similar if all their features except for sex are the same. Thus the data producer maps two individuals that differ only in their sex closely to each other in latent space. Consequently, the data consumer cannot infer the gender from the latent representation.

**tl;dr** yes, this is intended as the data producer has access to all features.
